cunumeric/module.py

```diff
@@ -1220,6 +1220,27 @@ def __init__(self, ndim, shape, dtype):
         self.dtype = dtype
 
 
+def check_list_depth(active_arr, leaf_map):
+    for each in active_arr:
+        if isinstance(each[1], list):
+            nlist = 0
+            for arr in each[1]:
+                if isinstance(arr, list):
+                    # check the depth of elements in this subarray
+                    check_list_depth([((*each[0], nlist), arr)], leaf_map)
+                    nlist += 1
+            # this is a leaf
+            if nlist == 0:
+                if len(each[1]) == 0:
+                    raise ValueError(f"List at {each[0]} " f"cannot be empty")
+                leaf_map[len(each[0])].append((each[0], each[1]))
+            elif nlist < len(each[1]):
+                raise ValueError(
+                    f"some elements in depth {len(each[0])} "
+                    f"have different depths"
```

Comment:
Don't use an f-string unless you have an expression in it. I'm surprised that the pre-commit hook even allowed you to commit this change.